tests/unit/core/test_pipeline.py

```diff
@@ -581,3 +584,28 @@ def test_refit(self, isinstance_mock):
         self.pipeline.refit(pd.Timestamp("2000.01.02"), pd.Timestamp("2022.01.02"))
 
         first_step.refit.assert_called_once_with(pd.Timestamp("2000.01.01"), pd.Timestamp("2022.01.01"))
+
+
+    def test_online_with_filled_buffer(self):
+
+        time = pd.date_range('2000-01-01', freq='1D', periods=7)
+        da = xr.DataArray([2, 3, 4, 3, 3, 1, 2], dims=["time"], coords={'time': time})
+        pipeline = Pipeline(batch=pd.Timedelta(days=1))
+        step_one = MagicMock()
+        step_one.get_result.return_value = {"step": da}
+
+        pipeline.start_steps["foo"] = StartStep("foo"), None
+        pipeline.start_steps["foo"][0].last = False
+        step_one.further_elements.side_effect = [True, True, True, True, False]
+        pipeline.add(module=step_one, input_ids=[1])
+
+        pipeline.test(pd.DataFrame({"foo": [1, 2, 2, 3, 4,5,6]},
+                                        index=pd.DatetimeIndex(pd.date_range('2000-01-01', freq='24H', periods=7))),
+                    online_start=pd.to_datetime('2000-01-04'))
```

Comment:
refactor code such that `index=...` aligns with `{"foo": ...` in line 602
and
`online_start=...` aligns with `pd.DataFrame(...` in line 602